examples/bzlmod/hello_world/third-party/extension.bzl

```diff
@@ -0,0 +1,14 @@
+"""Module extensions for using vendored crates with bzlmod"""
+
+load("//third-party/crates:defs.bzl", _crate_repositories = "crate_repositories")
+
+def _crate_repositories_impl(module_ctx):
+    direct_deps = _crate_repositories()
+    return module_ctx.extension_metadata(
+        root_module_direct_deps = [repo.repo for repo in direct_deps],
+        root_module_direct_dev_deps = [],
+    )
+
+crate_repositories = module_extension(
+    implementation = _crate_repositories_impl,
+)
```

Comment:
Separately, we should make crate_universe automatically generate a suitable module extension. Nothing in this file is project-specific.

Reply:
I think https://github.com/bazelbuild/rules_rust/issues/2181 is probably the umbrella issue for this... cc @matts1